app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt

```diff
@@ -51,13 +53,31 @@ open class TwoPaneFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         setupSlidingPane()
+        observeFolderName()
         observeThreadEvents()
     }
 
     private fun setupSlidingPane() {
         slidingPaneLayout.lockMode = SlidingPaneLayout.LOCK_MODE_LOCKED
     }
 
+    private fun observeFolderName() = with(mainViewModel) {
+
+        Utils.waitInitMediator(currentFolder, isInSearch).observe(viewLifecycleOwner) { (folder, isInSearch) ->
+
+            val displayedFolder = (if (isInSearch) {
+                Folder().apply { name = getString(R.string.searchFolderName) }
+            } else {
+                folder
+            }) ?: return@observe
+
+            rightPaneFolderName.value = getString(
+                R.string.noConversationSelected,
+                displayedFolder.getLocalizedName(requireContext()),
+            )
+        }
+    }
```

Comment:
You shouldn't expose `isInSearch` and instead you can have a `LiveData` returning the folder name. That way, all the logic of which name you have to display is only in the `ViewModel`.